lib/components/modal.vue

```diff
@@ -69,20 +69,32 @@
         opacity: 0 !important;
     }
 
-    /* Make modal display as block instead of inline style, and because Vue's v-show deletes inline "display" style*/
+    /* Make modal display as block instead of inline style, and because Vue's v-show deletes inline "display" style */
     .modal {
         display: block;
     }
 </style>
 
 <script>
+    import generateId from '../mixins/generate-id';
     import bBtn from './button.vue';
 
+    const SELECTOR = [
+        'button:not([disabled])',
+        'input:not([disabled])',
+        'select:not([disabled])',
+        'textarea:not([disabled])',
+        'a:not([disabled]):not(.disabled)',
+        '[tabindex]:not([disabled]):not(.disabled)'
+    ].join(',');
+
     export default {
+        mixins: [generateId],
```

Comment:
Thanks @tmorehouse that's so sad that we currently can't have such behavior. I'll make an on/off switch for `generateId`s so non-ssr users can use them optionally until we can find better way generating consistent IDs :)